doc/Learn/Navigation/HowTo-DefineRoutes.md

```diff
@@ -0,0 +1,326 @@
+
+```csharp
+protected override void RegisterRoutes(IViewRegistry views, IRouteRegistry routes)
+{
+    views.Register(
+        new ViewMap(ViewModel: typeof(ShellViewModel)),
+        new ViewMap<LoginPage, LoginViewModel>(),
+        new ViewMap<MainPage, MainViewModel>()
+    );
+
+    routes.Register(
+        new RouteMap("", View: views.FindByViewModel<ShellViewModel>(),
+            Nested:
+            [
+                new ("Login", View: views.FindByViewModel<LoginViewModel>()),
+                new ("Main", View: views.FindByViewModel<MainViewModel>(),
+                    Nested:
+                    [
+                        new ("ForYouTab"),
+                        new ("FavoritesTab")
+                    ]
+                )
+            ]
+        )
+    );
+}
+```
+
+#### IsDefault
+
+**IsDefault** will make the navigator automatically shows that route when dealing with nested views. For example, imagine a scenario where you are defining routes within a `NavigationView`, such as nested routes inside a `MainPage`. You can set one of these routes as the default by setting `IsDefault: true`. This ensures that the specified route is automatically navigated to when the page is displayed.
+
+```csharp
+// Omitted for brevity
+new RouteMap("Main", View: views.FindByViewModel<MainViewModel>(),
+    Nested:
+    [
+        new ("ForYouTab", IsDefault: true),
+        new ("FavoritesTab")
+    ]
+)
+```
+
+#### DependsOn
+
+**DependsOn** enables you to establish a dependency between two views. This argument expects a route name and ensures that when you navigate to a view with dependencies, the dependent view will be navigated to first before opening the requested view. This is especially useful when using deep linking to navigate through pages. In the following example we add a new page called `ProductsPage` and we set this page as dependent of `MainPage`:
+
+```csharp
+protected override void RegisterRoutes(IViewRegistry views, IRouteRegistry routes)
+{
+    views.Register(
+        // Omitted for brevity
+        new ViewMap<ProductsPage, ProductsViewModel>()
+    );
+
+    routes
+        .Register(
+            new RouteMap("", View: views.FindByViewModel<ShellViewModel>(),
+                Nested:
+                [
+                    // Omitted for brevity
+                    new ("Main", View: views.FindByViewModel<MainViewModel>()),
+                    new ("Products", View: views.FindByViewModel<ProductsViewModel>(), DependsOn: "Main"),
+                ]
+            )
+        );
+}
+```
+
+#### Init
+
+**Init** enables you to override the navigation request, allowing for custom initialization actions before displaying the associated view.
+
+```csharp
+new ("Main", View: views.FindByViewModel<MainViewModel>(),
+    Init: (request) => 
+    {
+        if(!User.IsLoggedIn())
+        {
+            request = request with { Route = Route.PageRoute("Login") };
```

Comment:
We should maybe add an explanation of what is going on here. This would be overriding the route of the original request and redirecting to the Login route?

Reply:
Yes. I added a few comments, let me know if it makes sense.